<launch>        
  
  <param name="/tag_type" value="16h5" />
<!-- 
   <include file="$(find video_stream_opencv)/launch/camera.launch" >
	  	<arg name="camera_name" value="camera" />
	  	<arg name="video_stream_provider" value="0" />
	  	<arg name="fps" value="30" />
	  	<arg name="frame_id" value="camera" />
	  	<arg name="camera_info_url" value="file:///$(find video_stream_opencv)/config/camera.yaml" />
	  	<arg name="flip_horizontal" value="false" />
	  	<arg name="flip_vertical" value="false" />
	  	<arg name="visualize" value="true" />
   </include>
 -->
<!-- 
  <node pkg="dji_sdk_read_cam" type="dji_sdk_read_cam" name="dji_sdk_read_cam" output="screen">
    <param name="gray_or_rgb" type="int" value="1"/>
    <param name="to_mobile" type="int" value="1"/>
  </node>
 -->
  <node pkg="apriltags" type="apriltags" name="apriltags" output="screen">
    <!-- <param name="camera_info_url" value="file:///$(find video_stream_opencv)/config/camera.yaml" /> -->
    <param name="~default_tag_size" value="0.06" />

    <param name="~viewer" value="true" />
    <param name="~publish_detections_image" value="false" />

    <param name="~display_type" value="ARROW" />
    <param name="~marker_thickness" value="0.02" />
    <param name="~display_marker_axes" value="true" />
    <param name="~display_marker_id" value="true" />
    <param name="~display_marker_overlay" value="false" />

    <rosparam command="delete" param="tag_data" />
    <rosparam param="tag_data">
      "2": 
        size: 0.06
      "6":
        size: 0.39
    </rosparam>

    <remap from="~image" to="/dji_sdk/image_raw"/>
    <remap from="~camera_info" to="/dji_sdk/camera_info"/>
    <remap from="~marker_array" to="/apriltags/marker_array"/>
    <remap from="~detections" to="/apriltags/detections"/>
  </node>

 <!--  <node name="tag_view" pkg="image_view" type="image_view">
    <remap from="image" to="/apriltags/detections_image" />
    <param name="autosize" type="bool" value="true"/>
  </node> -->

  <node name="controller_node" pkg="apriltags" type="controller_node" output="screen">
    <rosparam command="load" file="$(find apriltags)/resource/controller.yaml" />
  </node>

</launch>
